tools/engine_tool/lib/src/build_utils.dart

```diff
@@ -179,13 +182,24 @@ Future<int> runBuild(
   return buildResult ? 0 : 1;
 }
 
-/// Given a [Build] object, run only its GN step.
-Future<int> runGn(
+/// Run a [build]'s GN step if the output directory is missing.
+Future<bool> ensureBuildDir(
```

Comment:
I'd suggest keeping `runGn` and maybe making it lib private (`_runGn`), and then implementing `ensureBuildDir()` as a public wrapper around it.

Reply:
Done!